frontend/src/components/modals/settings/AutocompleteCombobox.tsx

```diff
@@ -58,16 +67,21 @@ export function AutocompleteCombobox({
         label={t(LABELS[ariaLabel])}
         placeholder={t(PLACEHOLDERS[ariaLabel])}
         defaultItems={items}
-        inputValue={
-          // Find the label for the default key, otherwise use the default key itself
-          // This is useful when the default key is not in the list of items, in the case of a custom LLM model
-          items.find((item) => item.value === defaultKey)?.label || defaultKey
-        }
```

Comment:
When `allowCustomValue` is true, the input field is pre-populated with the `defaultKey`, and it gets updated based on the user's input via the `onChange` method called on `onInputChange`. Meanwhile, when `allowCustomValue` is false, we don't update the defaultKey via `onChange` on every `onInputChange`. Instead, we only call `onChange` when the selection is changed (`onSelectionChange`) from the dropdown list. 

It makes more sense not to call the `onChange` method with every input change for the field where the user should select from the dropdown, and custom values are not allowed.